modules/ldap/src/main/java/com/nhds/wasabi/ldap/impl/DirectoryAuthentication.java

```
@@ -0,0 +1,184 @@
+        if (!authHeader.isPresent()) {
+            throw new AuthenticationException("Null Authentication Header is not supported");
+        }
+
+        if (!authHeader.or(SPACE).contains(BASIC)) {
+            throw new AuthenticationException("Only Basic Authentication is supported");
+        }
+
+        final String encodedUserPassword = authHeader.get().substring(authHeader.get().lastIndexOf(SPACE));
+        String usernameAndPassword;
+
+        LOGGER.trace("Base64 decoded username and password is: {}", encodedUserPassword);
+
+        try {
+            usernameAndPassword = new String(decodeBase64(encodedUserPassword.getBytes()));
+        } catch (Exception e) {
+            throw new AuthenticationException("error parsing username and password", e);
+        }
+
+        String[] fields = usernameAndPassword.split(SEMICOLON);
+
+        if (fields.length > 2) {
+            throw new AuthenticationException("More than one username and password provided, or one contains ':'");
+        } else if (fields.length < 2) {
+            throw new AuthenticationException("Username or password are empty.");
+        }
+
+        if (isBlank(fields[0]) || isBlank(fields[1])) {
+            throw new AuthenticationException("Username or password are empty.");
+        }
+
+        return new DirectoryUserCredential(fields[0], fields[1]);
+    }
+
+    /**
+     * Attempts to verify the user token retrieved via the {@link #logIn(String) logIn} method.
+     *
+     * @param tokenHeader the token header
+     * @return a login token for this user (always)
+     * @throws AuthenticationException for invalid tokens
+     */
+    @Override
+    public LoginToken verifyToken(final String tokenHeader) {
+        LOGGER.debug("Authentication token received as: {}", tokenHeader);
+
+        DirectoryUserCredential credential = parseUsernamePassword(fromNullable(tokenHeader));
+
+        if (userDirectory.isDirectoryTokenValid(credential.username, credential.password)) {
```

Same as above. You can just throw the exception if not valid and then return it without the else.